internal/handlers/response.go

```diff
@@ -11,6 +11,44 @@ import (
 	"github.com/authelia/authelia/internal/utils"
 )
 
+// HandleOIDCWorkflowResponse handle the redirection upon authentication in the OIDC workflow.
+func HandleOIDCWorkflowResponse(ctx *middlewares.AutheliaCtx) {
+	userSession := ctx.GetSession()
+
+	if !authorization.IsAuthLevelSufficient(userSession.AuthenticationLevel, userSession.OIDCWorkflowSession.RequiredAuthorizationLevel) {
+		ctx.Logger.Warn("OIDC requires 2FA, cannot be redirected yet")
+		ctx.ReplyOK()
+
+		return
+	}
+
+	uri, err := ctx.ForwardedProtoHost()
+	if err != nil {
+		ctx.Logger.Errorf("%v", err)
+		handleAuthenticationUnauthorized(ctx, fmt.Errorf("Unable to get forward facing URI"), authenticationFailedMessage)
+
+		return
+	}
+
+	isConsentMissingScopes := len(userSession.OIDCWorkflowSession.RequestedScopes) > 0 && (userSession.OIDCWorkflowSession == nil ||
```

Comment:
should it simply be
```
(userSession.OIDCWorkflowSession != nil &&
		utils.IsStringSlicesDifferent(userSession.OIDCWorkflowSession.RequestedScopes, userSession.OIDCWorkflowSession.GrantedScopes))
```

Reply:
Same for the other condition below?

Reply:
I have factorized these so they are a function now since they're reused. Also I think that's what was originally there, but I think we should carefully think about the logic of whatever we put in there. We should err on the side of caution and assume scopes are missing unless we're certain.

I have a feeling we should be rejecting anything requesting zero scopes anyway.